db-server/test/backend/db_tests.js

```diff
@@ -2058,11 +2058,16 @@ module.exports = function (config, DB) {
       })
 
       const codeLengthTest = [0, 4, 8]
+      const codeTest = /^[a-z0-9]{10}$/
       codeLengthTest.forEach((num) => {
         it('should generate ' + num + ' recovery codes', () => {
           return db.replaceRecoveryCodes(account.uid, num)
             .then((codes) => {
               assert.equal(codes.length, num, 'correct number of codes')
+              codes.forEach((code) => {
+                assert.equal(codeTest.test(code), true, 'matches recovery code format')
+              })
+              assert.equal()
```

Comment:
Not commenting on this part of the diff, but about the tests for `replaceRecoveryCodes` underneath. Should we add some coverage that the new `.salt` property seems right? And I don't see anything for `codeHash` either fwiw.

Reply:
This was added in the `dbUtils` test module.